assignment.py

```diff
@@ -0,0 +1,74 @@
+from decorators import login_required
+from db import db, User, Course, PreferenceAssignment, Teacher, Researcher, Organization, PublishAssignment, \
+    ResearcherSupervisor
+from flask import Blueprint, render_template, flash, current_app, url_for, request, make_response, redirect, session, \
+    Flask, jsonify
+from util import get_current_year
+
+assignment_bp = Blueprint('assignment', __name__)
+
+
+@assignment_bp.route('/assignments', methods=['GET'])
+@login_required
+def assignments():
+    return render_template('assignment.html')
+
+
+def serialize_model(model):
+    """Converts a SQLAlchemy model object into a dictionary."""
+    return {column.name: getattr(model, column.name) for column in model.__table__.columns}
+
+
+@assignment_bp.route('/load_data', methods=['GET'])
+@login_required
+def load_data():
+    current_year = get_current_year()
+    courses = (db.session.query(Course).filter_by(year=current_year).order_by(Course.quadri).all())
+    users = db.session.query(User).filter_by(active=True).all()
```

Comment:
Beware: if you save assignments and then de-activate a user, their data won't be found when reloading the assignment table. You should check for this.